spec.html

```diff
@@ -27849,123 +28115,85 @@ <h1>
+          <dd></dd>
         </dl>
-        <p>An implementation of HostImportModuleDynamically must conform to the following requirements:</p>
 
-        <ul>
-          <li>
-            It must return ~unused~. Success or failure must instead be signaled as discussed below.
-          </li>
-          <li>
-            The host environment must conform to one of the two following sets of requirements:
-            <dl>
-              <dt>Success path</dt>
-
-              <dd>
-                <ul>
-                  <li>At some future time, the host environment must perform FinishDynamicImport(_referencingScriptOrModule_, _specifier_, _promiseCapability_, _promise_), where _promise_ is a Promise resolved with *undefined*.</li>
+        <emu-note>
+          <p>An example of when _referrer_ can be a Realm Record is in a web browser host. There, if a user clicks on a control given by</p>
 
-                  <li>Any subsequent call to HostResolveImportedModule after FinishDynamicImport has completed, given the arguments _referencingScriptOrModule_ and _specifier_, must return a normal completion containing a module which has already been evaluated, i.e. whose Evaluate concrete method has already been called and returned a normal completion.</li>
-                </ul>
-              </dd>
+          <pre><code class="html">&lt;button type="button" onclick="import('./foo.mjs')"&gt;Click me&lt;/button&gt;</code></pre>
 
-              <dt>Failure path</dt>
+          <p>there will be no active script or module at the time the <emu-xref href="#sec-import-calls">`import()`</emu-xref> expression runs. More generally, this can happen in any situation where the host pushes execution contexts with *null* ScriptOrModule components onto the execution context stack.</p>
+        </emu-note>
 
-              <dd>
-                <ul>
-                  <li>At some future time, the host environment must perform FinishDynamicImport(_referencingScriptOrModule_, _specifier_, _promiseCapability_, _promise_), where _promise_ is a Promise rejected with an error representing the cause of failure.</li>
-                </ul>
-              </dd>
-            </dl>
+        <p>An implementation of HostLoadImportedModule must conform to the following requirements:</p>
+        <ul>
+          <li>
+            The host environment must perform FinishLoadingImportedModule(_referrer_, _specifier_, _payload_, _result_), where _result_ is either a normal completion containing the loaded Module Record or a throw completion, either synchronously or asynchronously.
```

Comment:
In the case where a module has already been loaded, is it up to the host whether it resolves synchronously or asynchronously? I wonder if we should mandate already-loaded modules return synchronously to avoid unnecessary divergence?

Reply:
Until last week HTML was ambiguous allowed loading of already-loaded modules to be asynchronous. I updated HTML to make it synchronous, but without first discussing with implementers, and browsers don't currently agree.

I'm working on it on https://github.com/web-platform-tests/wpt/pull/35949, but I prefer to keep it out of this PR (since the consensus was based on the fact that this was not observably normative in common hosts).

Reply:
Glad to hear you're working on it!